src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -12,109 +12,119 @@ import ReportActionPropTypes from './ReportActionPropTypes';
-    {
-        text: 'Mark as Unread',
-        icon: Mail,
-        shouldShow: false,
-        onPress: () => {},
-    },
-
-    // Edit Comment
-    {
-        text: 'Edit Comment',
-        icon: Pencil,
-        shouldShow: false,
-        onPress: () => {},
-    },
-
-    // Delete Comment
-    {
-        text: 'Delete Comment',
-        icon: Trashcan,
-        shouldShow: false,
-        onPress: () => {},
-    },
-];
-
 const propTypes = {
     // The ID of the report this report action is attached to.
     // eslint-disable-next-line react/no-unused-prop-types
     reportID: PropTypes.number.isRequired,
 
     // The report action this context menu is attached to.
-    reportAction: PropTypes.shape(ReportActionPropTypes),
+    reportAction: PropTypes.shape(ReportActionPropTypes).isRequired,
 
     // If true, this component will be a small, row-oriented menu that displays icons but not text.
     // If false, this component will be a larger, column-oriented menu that displays icons alongside text in each row.
     isMini: PropTypes.bool,
 
     // Controls the visibility of this component.
     isVisible: PropTypes.bool,
+
+    // Function to trigger when we try to mark a message as unread
+    onMarkAsUnread: PropTypes.func,
 };
 
 const defaultProps = {
-    reportAction: {},
     isMini: false,
     isVisible: false,
+    onMarkAsUnread: () => {},
 };
 
-const ReportActionContextMenu = (props) => {
-    const wrapperStyle = getReportActionContextMenuStyles(props.isMini);
-    return props.isVisible && (
-        <View style={wrapperStyle}>
-            {CONTEXT_ACTIONS.map(contextAction => contextAction.shouldShow && (
-                <ReportActionContextMenuItem
-                    icon={contextAction.icon}
-                    text={contextAction.text}
-                    successIcon={contextAction.successIcon}
-                    successText={contextAction.successText}
-                    isMini={props.isMini}
-                    onPress={() => contextAction.onPress(props.reportAction)}
-                    key={contextAction.text}
-                />
-            ))}
-        </View>
-    );
-};
+class ReportActionContextMenu extends React.Component {
+    constructor(props) {
+        super(props);
+
+        /**
+         * A list of all the context actions in this menu.
+         */
+        this.CONTEXT_ACTIONS = [
+            // Copy to clipboard
+            {
+                text: 'Copy to Clipboard',
+                icon: ClipboardIcon,
+                successText: 'Copied!',
+                successIcon: Checkmark,
+                shouldShow: true,
+
+                // If return value is true, we switch the `text` and `icon` on
+                // `ReportActionContextMenuItem` with `successText` and `successIcon` which will fallback to
+                // the `text` and `icon`
+                onPress: () => {
+                    const message = _.last(lodashGet(this.props.reportAction, 'message', null));
```

Comment:
I feel like this should be a class method so that it prevents this constructor from turning into a monolithic block of logic and functions.

Reply:
I see this is being discussed in the `delete comments` PR, so I'll wait for the resolution over there